### ISSUE TITLE
Incorrect error on runtime missing with firebase deploy with Python cloud functions

### ISSUE THREAD

**Issue by JoseCSantos** [firebase/firebase-tools#7559]
### [REQUIRED] Environment info

**firebase-tools:** 13.15.1

**Platform:** Windows 11

### [REQUIRED] Test case

**main.py**
from firebase_admin import initialize_app, firestore, db, credentials

# Initialize Firebase Admin SDK
credentials_filename = 'my-firebase-credentials.json'
db_url = 'https://my_db.firebasedatabase.app'
cred = credentials.Certificate(credentials_filename)
initialize_app(credential=cred, options={'databaseURL': db_url})

**firebase.json**
{
  "functions": 
    {
      "source": "functions",
      "codebase": "default"
    }
}

**package.json**
{
  "name": "functions",
  "version": "1.0.0",
  "description": "Testing cloud functions",
  "main": "main.py",
  "scripts": {
  },
  "engines" : {
      "python": "python311"
    }
}

### [REQUIRED] Steps to reproduce

Run 'firebase deploy --only functions'

with the above files.
File main.py at functions\main.py
File package.json at functions\package.json
File firebase.json at .\firebase.json

It's important to note that Python cloud functions work on the emulator through firebase emulators:start
Have tried removing engines and add 'engine' and don't add engines/engine on package.json and do 'runtime': python311 on firebase.json but always get the same error (see actual behaviour below).

When an invalid runtime is specified, such as python39, I get error:
Error: Functions codebase default has invalid runtime python39 specified in firebase.json. Valid values are:
- nodejs6
- nodejs8
- nodejs10
- nodejs12
- nodejs14
- nodejs16
- nodejs18
- nodejs20
- nodejs22
- python310
- python311
- python312

which hints that python311 would be supported and recognized.

Issue is similar to https://stackoverflow.com/questions/78433681/firebase-deploy-runtime-field-not-found-in-firebase-json-or-package-json from May 2024 which got no response yet.

### [REQUIRED] Expected behavior

Cloud functions deployed with no error.

### [REQUIRED] Actual behavior

Error: `runtime` field is required but was not found in firebase.json or package.json.
To fix this, add the following lines to the `functions` section of your firebase.json:
"runtime": "nodejs22" or set the "engine" field in package.json

**Comment by aalej** [comment]
Hey @JoseCSantos, sorry to hear that you encountered this issue, and thanks for reaching out to us. Based on the information you provided, I think there might be a configuration issue here. The `package.json` is used for determining the [Node.js runtime](https://firebase.google.com/docs/functions/manage-functions?gen=2nd#set_nodejs_version
), but when using Python, you can [set the runtime](https://firebase.google.com/docs/functions/manage-functions?gen=2nd#set_python_version
) in the `firebase.json` file.

Could you try:
1. Removing the `functions/package.json` file as the runtime set here might be conflicting with the one set in `firebase.json`
2. Setting the runtime in your `firebase.json` file:
```json
{
 "functions": {
   "source": "functions",
   "codebase": "default",
   "runtime": "python311"
 }
}
```
3. Run `firebase deploy --only functions`

Are you able to deploy the functions after applying the changes above? If not, any chance you could share an [MCVE](https://stackoverflow.com/help/minimal-reproducible-example) to help us narrow down the cause of the issue?

**Comment by JoseCSantos** [comment]
Hi @aalej,

Thank you very much for the prompt response.
I've removed the functions/package.json as you suggested and it worked, the functions deployed!

Could I suggest that, if the source is Python, firebase doesn't create the package.json file ? That would mitigate this issue.

Thanks again,

José

**Comment by aalej** [comment]
Awesome! It's good to hear that you were able to get the deployments working. Just to verify, could you try running `firebase init functions` in a new directory then selecting `Python` to see if a `functions/package.json` file is being generated?

If a `functions/package.json` file is generated, this is definitely something we should look into as this would cause issues similar to what you encountered.  

I tried this out on my end and the `functions/package.json` wasn't generated. I'm using a macOS machine so I'm not sure if the behavior could be different for Windows.

**Comment by JoseCSantos** [comment]
Hi @aalej,

Just tried in a new directory and indeed `functions\package.json` was **not** generated when I select Python.

The reason I had the package.json file is because I chose Javascript when I ran `firebase init functions` the first time but then decided to go with Python, run  `firebase init functions`  and the package.json file still lingered there.

This scenario where a dev first picks JS/TS and then changes to Python on the same directory may not be that uncommon.
To make firebase deploy robust to this scenario I would suggest to, if the runtime is Python, ignore the package.json file if present.

If that's complicated, updating the error message to mention that, when using Python runtime, package.json should be manually deleted, would also mitigate this issue.

Thanks again for your help.
